x/concentrated-liquidity/concentrated-pool/pool.go

```diff
@@ -0,0 +1,101 @@
+package concentrated_pool
+
+import (
+	"encoding/json"
+	fmt "fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v12/x/concentrated-liquidity/types"
+	gammtypes "github.com/osmosis-labs/osmosis/v12/x/gamm/types"
+)
+
+var (
+	_ types.PoolI = &Pool{}
+)
+
+func NewConcentratedLiquidityPool(poolId uint64, denom0, denom1 string, currSqrtPrice sdk.Dec, currTick sdk.Int) (Pool, error) {
+	denom0, denom1, err := types.OrderInitialPoolDenoms(denom0, denom1)
+	if err != nil {
+		return Pool{}, err
+	}
+	pool := Pool{
+		// TODO: move gammtypes.NewPoolAddress(poolId) to swaproutertypes
```

Comment:
why? I think it belongs here

Reply:
Isn't the purpose of swaprouter types be the module of orchestration between pool models? 
I think at least we should create a method for newpooAddress within the concentrated liquidity module rather than creating a dependecy flow between concenterated liquidity package and the gamm types package. Wdyt?

Reply:
ahh I see, that makes sense. I thought you meant to move address creation itself there and propagate the address from the swap router.

I'm down for your suggestion